pinot-common/src/main/java/com/linkedin/pinot/common/utils/DataTable.java

```diff
@@ -30,38 +33,60 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Map.Entry;
-
 import org.apache.commons.io.IOUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.linkedin.pinot.common.Utils;
-import com.linkedin.pinot.common.data.FieldSpec.DataType;
-import com.linkedin.pinot.common.response.ProcessingException;
-import com.linkedin.pinot.common.utils.DataTableBuilder.DataSchema;
-
 
 /**
  *
  * Read only Datatable. Use DataTableBuilder to build the data table
  */
 public class DataTable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTable.class);
 
   public static final String EXCEPTION_METADATA_KEY = "Exception";
-
   private static final Charset UTF8 = Charset.forName("UTF-8");
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(DataTable.class);
+  // Data Table version
+  public enum Version {
+    JavaSerialized(1), // Keep the value of '1' for backward compatibility
+    CustomSerialized(2);
+
+    private int value;
+    private static Map<Integer, Version> map = new HashMap<Integer, Version>();
```

Comment:
Actually, the reason map is required is because I am keeping the JavaSerialized ordinal as 1 instead of zero. Doing so allows to keep backward compatibility.